src/containers/BookForm.js

```diff
@@ -1,17 +1,76 @@
-import React from 'react';
+import React, { useState } from 'react';
+import propTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createBook } from '../actions';
+
+function BookForm(props) {
+  const [book, setBook] = useState({
+    title: '',
+    category: 'Biography',
+  });
+
+  const handleChange = e => {
+    switch (e.target.id) {
+      case 'title':
+        setBook({
+          ...book,
+          title: e.target.value,
+        });
+        break;
+
+      case 'category':
+        setBook({
+          ...book,
+          category: e.target.value,
+        });
+        break;
+
+      default:
+        break;
+    }
+  };
```

Comment:
- [ ] [OPTIONAL] This logic works fine, but since you gave each input the same id as the name of the state field to be changed you can write this in a much more simple way:

Ex:
```
  const handleChange = e => {
        setBook({
          ...book,
          [e.target.id]: e.target.value,
        });
  };
```